scripts/src/java/org/oppia/android/scripts/maven/GenerateLicenseTexts.kt

```diff
@@ -0,0 +1,423 @@
+    stringElement.appendChild(doc.createTextNode(version))
+    rootResourcesElement.appendChild(stringElement)
+  }
+
+  // Add an array of dependencies versions string resources.
+  val stringArrayElement = doc.createElement("string-array")
+  stringArrayElement.setAttribute("name", "third_party_dependencies_versions_array")
+  for (index in dependencyVersionsList.indices) {
+    val itemElement = doc.createElement("item")
+    itemElement.appendChild(doc.createTextNode("@string/third_party_dependency_version_$index"))
+    stringArrayElement.appendChild(itemElement)
+  }
+
+  rootResourcesElement.appendChild(stringArrayElement)
+  doc.appendChild(rootResourcesElement)
+  doc.xmlStandalone = true
+  getTransformer().transform(DOMSource(doc), StreamResult(file))
+}
+
+private fun writeDependenciesLicensesXml(
+  pathToResourceXml: String,
+  copyrightLicenseSet: Set<CopyrightLicense>
+): HashMap<String, String> {
+  val file = File(pathToResourceXml)
+  val licenseMap = hashMapOf<String, String>()
+  val docBuilder: DocumentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+  val doc: Document = docBuilder.newDocument()
+  val rootResourcesElement: Element = doc.createElement("resources")
+
+  // Add all license texts.
+  for (index in copyrightLicenseSet.indices) {
+    val licenseLink = copyrightLicenseSet.elementAt(index).licenseLink
+    val licenseText = "<![CDATA[${copyrightLicenseSet.elementAt(index).licenseText}]]>"
+    val stringElement = doc.createElement("string")
+    stringElement.setAttribute("name", "license_$index")
+    stringElement.appendChild(doc.createTextNode(licenseText))
+    rootResourcesElement.appendChild(stringElement)
+    licenseMap[licenseLink] = "license_$index"
+  }
+
+  doc.appendChild(rootResourcesElement)
+  doc.xmlStandalone = true
+  getTransformer().transform(DOMSource(doc), StreamResult(file))
+  return licenseMap
+}
+
+private fun writeDependenciesLicenseTextsArray(
+  pathToResourceXml: String,
+  licenseLinkToIndexNameMap: HashMap<String, String>,
+  mavenDependenciesList: List<MavenDependency>
+) {
+  val file = File(pathToResourceXml)
+  val docBuilder: DocumentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+  val doc: Document = docBuilder.newDocument()
+  val rootResourcesElement: Element = doc.createElement("resources")
+
+  // Add all string-arrays of license texts for each dependency.
+  for (index in mavenDependenciesList.indices) {
+    val dependency = mavenDependenciesList[index]
+    val stringArrayElement = doc.createElement("string-array")
+    stringArrayElement.setAttribute("name", "third_party_dependency_license_texts_$index")
+    dependency.licenseList.forEach { license ->
+      val licenseLink = if (license.primaryLinkType == PrimaryLinkType.SCRAPE_DIRECTLY ||
+        license.primaryLinkType == PrimaryLinkType.SHOW_LINK_ONLY
+      ) {
+        license.primaryLink
+      } else {
+        license.alternativeLink
+      }
```

Comment:
This still isn't making sense to me--why would we keep a link that can't be shown? What's the case where we want to show the primary but not the secondary, or vice versa?

If the purpose is to retain the original link and allow for a local copy, then maybe using a ``oneof`` would be a better approach:

```proto
oneof license_link_type {
  ScrapableLink scrapable_link = 1;
  LocalCopy local_copy = 2;
  LinkOnly link_only = 3;
}

message ScrapableLink {
  string url = 1;
}

message LocalCopy {
  string filename = 1;
}

message DirectLink {
  string url = 1;
}
```

Reply:
Done.